streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java

```diff
@@ -64,13 +64,14 @@
      * query the value of the key on a parallel running instance of your Kafka Streams application.
      * <p>
      * For failure and recovery the store will be backed by an internal changelog topic that will be created in Kafka.
+     * Therefore, the store name must be a valid Kafka topic name and cannot contain characters other than ASCII alphanumerics, '.', '_' and '-'.
```

Comment:
Nit: we try to follow the following policy to make reviewing easier:
 - each new sentence start a new line (you do follow this already)
 - no line longer then 120 character (otherwise lines are longer then Github displays and you need to scroll to the right)